src/ng/directive/input.js

```diff
@@ -1476,6 +1476,8 @@ function radioInputType(scope, element, attr, ctrl) {
 
   ctrl.$render = function() {
     var value = attr.value;
+    // Strict comparison would cause a BC
```

Comment:
I've added a test here: https://github.com/angular/angular.js/pull/14313/files#diff-ce6d0f141f60d25eede7abf044d04b93 Without strict comparison, you can set your model to `0` and it will still check a radio button that has `value="0"`(string)